src/cpp/DQM/serverext.cc

```diff
@@ -2341,6 +2341,22 @@ class VisDQMToJSON
 	  }
 	}
 
+	// Match run numbers against run regexp if one was given.
+	if (options.count("run"))
+	{
+	  std::string rxerr;
+	  makerx(options["run"], rx, rxerr, Regexp::IgnoreCase);
+	  if (rx)
+	  {
+	    VisDQMSamples final;
+	    final.reserve(samples.size());
+	    for (size_t i = 0, e = samples.size(); i != e; ++i)
+	      if (rx->search(std::to_string(samples[i].runnr)) >= 0)
+		final.push_back(samples[i]);
+	    std::swap(final, samples);
+	  }
+	}
+
 	for (size_t i = 0, e = samples.size(); i != e; ++i)
```

Comment:
Ciao @schneiml , thanks for this.
Should we, maybe, embed the `if` logic inside so that we avoid almost 100% code duplication?
Also, what will happen if the use supply both filters?

Reply:
Deduplicating the logic sure makes sense, though for the beginning I wanted to keep things simple. I can give it a try.

Using both filters does an _and_ which makes sense to me and works correctly.